File: js/packages/app/src/__tests__/features/object-references/internal/polywrap/internal-references-example/wrap/ImportBindings.ts
import { TestInternalClass, TestInternalClassImport } from "../external/classes";


export class ImportBindings {
  constructor(
    public testReturnReference: (
      arg: string
    ) => Promise<TestInternalClass>,
    public TestInternalClass: TestInternalClassImport,
    public TestObjectGetter: {
      new(arg: string): TestInternalClass;
    }
  ) {
  }
}

File: rust/app/src/wrappers/external_references/src/polywrap/external/module/import_bindings.rs

use crate::polywrap::external::{classes::test_external_class::TestExternalClassImport, global_functions::test_external_global_function::WrappedClosure};


pub struct ImportBindings {
    pub testExternalGlobalFunctionWrapped: WrappedClosure,
    pub TestExternalClass: TestExternalClassImport,
    
}

impl ImportBindings {
    pub fn new(
        test: WrappedClosure,
        TestExternalClass: TestExternalClassImport,
    ) -> Self {
        Self {
                                                                                                            
            testExternalGlobalFunctionWrapped: test,     
                                    
            TestExternalClass: TestExternalClass,
            
        }
    }

    pub fn testExternalGlobalFunction(&self, arg: String) -> String {
        self.testExternalGlobalFunctionWrapped.call(arg)
    }
}
